### DISCUSSION TITLE
Belchertown Skin v1.3.1

### DISCUSSION THREAD

**Discussion by poblabs** [poblabs/weewx-belchertown#885] [Announcements]
This is a minor release which pins HighCharts to version 10

---

**While unlikely something will go wrong, please make sure you have a backup your weewx.conf configuration.**

#### Download: 
`wget https://github.com/poblabs/weewx-belchertown/releases/download/weewx-belchertown-1.3.1/weewx-belchertown-release-1.3.1.tar.gz`

#### Then run:

`sudo wee_extension --install weewx-belchertown-release-1.3.1.tar.gz`

Configure the [weewx.conf pre-requisites](https://github.com/poblabs/weewx-belchertown#requirements) and restart weewx when done. 

<hr /><em>This discussion was created from the release <a href='https://github.com/poblabs/weewx-belchertown/releases/tag/weewx-belchertown-1.3.1'>Belchertown Skin v1.3.1</a>.</em>

**Comment by tomdotorg** [comment]
Download link is incorrect. Should be:

wget https://github.com/poblabs/weewx-belchertown/releases/download/weewx-belchertown-1.3.1/weewx-belchertown-release.1.3.1.tar.gz

**Reply by poblabs** [reply]
Updated it, thanks!

**Comment by lordratner** [comment]
File gets saved as `weewx-belchertown-release.1.3.1.tar.gz`

So this: `sudo wee_extension --install weewx-belchertown-release-1.3.1.tar.gz` 

needs to be this: `sudo wee_extension --install weewx-belchertown-release.1.3.1.tar.gz`

**Reply by czollars** [reply]
When I run the current skin install, it always comes back with a command not found.  It seems to download the file just fine.  Am I missing something?  I am trying to upgrade the skin to work with the new version of high charts.  I lost all my graphs.  

pi@raspberrypi:~ $ sudo wee_extension --install weewx-belchertown-release.1.3.1.tar.gz
sudo: wee_extension: command not found

**Reply by uajqq** [reply]
The install command has changed with recent versions of weewx. It's now `weectl extension install [...]`

**Reply by czollars** [reply]
weectl extension install weewx-belchertown-release.1.3.1.tar.gz

So it would look like this, for his latest skin file?

**Reply by uajqq** [reply]
Yes, that looks right.

**Reply by czollars** [reply]
Thanks, it is working.

**Comment by ericg75** [comment]
That cleared it up.  Resolved.  Thank you!

**Comment by vrishabkakade** [comment]
I seem to be getting this error when I tried to install v1.3.1  with the latest release of WeeWx - 5.0.2. I'm able to install v1.3 just fine on WeeWx 5.0.2. Am I doing anything wrong here?

vrishabkakade@freedompi:~ $ sudo weectl extension install weewx-belchertown-release.1.3.1.tar
Using configuration file /etc/weewx/weewx.conf
Install extension 'weewx-belchertown-release.1.3.1.tar' (y/n)? y
Traceback (most recent call last):
  File "/usr/share/weewx/weectl.py", line 74, in <module>
    main()
  File "/usr/share/weewx/weectl.py", line 66, in main
    namespace.func(namespace)
  File "/usr/share/weewx/weectllib/__init__.py", line 121, in dispatch
    namespace.action_func(config_dict, namespace)
  File "/usr/share/weewx/weectllib/extension_cmd.py", line 116, in install_extension
    ext.install_extension(namespace.source, no_confirm=namespace.yes)
  File "/usr/share/weewx/weecfg/extension.py", line 143, in install_extension
    raise InstallError(f"Unrecognized type for {extension_path}")
weecfg.extension.InstallError: Unrecognized type for weewx-belchertown-release.1.3.1.tar

**Reply by tomdotorg** [reply]
I am working on updating my Docker image for weewx5 and I want to put the extension in as a default. I am getting the same error as this. Has it been understood? Thanks.

**Reply by poblabs** [reply]
Is this because of the default database missing appTemp? In your Dockerfile try these commands before installing belchertown

https://github.com/poblabs/weewx-belchertown/issues/924#issuecomment-2048694009

```
weectl database add-column appTemp
weectl database drop-daily
weectl database rebuild-daily
```

**Comment by tomdotorg** [comment]
That column is in the default 5.0 db? In my Dockerfile, I have a database
create followed by an attempt to add that column, and that yields a
duplicate column error, so it looks to me like it is already there.

--
Tom Mitchell
***@***.***
Real Time Beverly, MA Weather
http://weather.tom.org

"The whole problem with the world
  is that fools and fanatics are
  always so certain of themselves,
  and wiser people so full of doubts."
  - Bertrand Russell

**Reply by poblabs** [reply]
Yeah I'm seeing that too on a new install. I'm looking into it as time allows (unless someone else has already solved it!)

**Comment by tomdotorg** [comment]
Thanks, Pat. I am blocked on this, but it is an enhancement and not urgent.

--
Tom Mitchell
***@***.***
Real Time Beverly, MA Weather
http://weather.tom.org

"The whole problem with the world
  is that fools and fanatics are
  always so certain of themselves,
  and wiser people so full of doubts."
  - Bertrand Russell

**Reply by poblabs** [reply]
When I try to install using the weewx5 git method, it's working fine for me. Are you using git or apt in your Dockerfile?

```
python3 ~/weewx/src/weectl.py extension install weewx-belchertown-release.1.3.1.tar.gz
Using configuration file /root/weewx-data/weewx.conf
Install extension 'weewx-belchertown-release.1.3.1.tar.gz' (y/n)? y
Extracting from tar archive weewx-belchertown-release.1.3.1.tar.gz
Saving installer file to /root/weewx-data/bin/user/installer/Belchertown
Saved copy of configuration as /root/weewx-data/weewx.conf.20240627150805
Finished installing extension Belchertown from weewx-belchertown-release.1.3.1.tar.gz
```

**Comment by tomdotorg** [comment]
I'm running the pip install:

RUN python3 -m pip install weewx --user --no-warn-script-location && \
    ~/.local/bin/weectl station create --no-prompt && \
    cat /tmp/logging-stdout.conf >> $WEEWX_ROOT/weewx.conf


 > [linux/arm64  8/11] RUN /home/weewx/.local/bin/weectl extension install
weewx-belchertown-release.1.3.1.tar.gz --yes --verbosity=3:
0.182 Using configuration file /home/weewx/weewx-data/weewx.conf
0.182 weectl[1] INFO weectllib: Initializing weectl version 5.0.2
0.182 weectl[1] INFO weectllib: Command line: /home/weewx/.local/bin/weectl
extension install weewx-belchertown-release.1.3.1.tar.gz --yes --verbosity=3
0.183 Traceback (most recent call last):
0.183   File "/home/weewx/.local/bin/weectl", line 8, in <module>
0.183     sys.exit(main())
0.183              ^^^^^^
0.183   File "/home/weewx/.local/lib/python3.11/site-packages/weectl.py",
line 66, in main
0.183     namespace.func(namespace)
0.183   File
"/home/weewx/.local/lib/python3.11/site-packages/weectllib/__init__.py",
line 121, in dispatch
0.183     namespace.action_func(config_dict, namespace)
0.183   File
"/home/weewx/.local/lib/python3.11/site-packages/weectllib/extension_cmd.py",
line 116, in install_extension
0.183     ext.install_extension(namespace.source, no_confirm=namespace.yes)
0.183   File
"/home/weewx/.local/lib/python3.11/site-packages/weecfg/extension.py", line
143, in install_extension
0.183     raise InstallError(f"Unrecognized type for {extension_path}")
0.183 weecfg.extension.InstallError: Unrecognized type for
weewx-belchertown-release.1.3.1.tar.gz
------

--
Tom Mitchell
***@***.***
Real Time Beverly, MA Weather
http://weather.tom.org

"The whole problem with the world
  is that fools and fanatics are
  always so certain of themselves,
  and wiser people so full of doubts."
  - Bertrand Russell

**Comment by tomdotorg** [comment]
FYI, I am seeing the same issue on install of the mqtt extension as well. I
can work around it by unzipping the distro and installing directly.

--
Tom Mitchell
***@***.***
Real Time Beverly, MA Weather
http://weather.tom.org

"The whole problem with the world
  is that fools and fanatics are
  always so certain of themselves,
  and wiser people so full of doubts."
  - Bertrand Russell

**Comment by Xanti19** [comment]
First of all, thank you for the great skin! The installation of version 1.3.1 also stopped for me yesterday because I copied the command with the dash in front of the 1. Please correct the instructions on the page accordingly so that the next ones don't have to overcome this hurdle too. ;-)

**Comment by wallaby-gopher** [comment]
I was getting an error:
```
File "/opt/venv/lib/python3.14/site-packages/weewx/cheetahgenerator.py", line 226, in generate

    ngen += self.generate(section[subsection], subsection, gen_ts)

            ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/opt/venv/lib/python3.14/site-packages/weewx/cheetahgenerator.py", line 309, in generate

    searchList = self._getSearchList(encoding, timespan,

                                     default_binding, section_name,

                                     os.path.join(

                                       os.path.dirname(report_dict['template']),

                                       _filename))

  File "/opt/venv/lib/python3.14/site-packages/weewx/cheetahgenerator.py", line 405, in _getSearchList

    search_list += obj.get_extension_list(timespan, db_lookup)

                   ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^

  File "/data/bin/user/belchertown.py", line 612, in get_extension_list

    locale.format("%g", float(year_outTemp_max_range_total)),

    ^^^^^^^^^^^^^

AttributeError: module 'locale' has no attribute 'format'
```
so I went to /data/bin/user/belchertown.py
and added the following below `import locale`
`locale.format = locale.format_string`

**Reply by uajqq** [reply]
This version of the skin is no longer being maintained. This issue has been fixed on the fork I'm maintaining, GitHub.com/uajqq/weewx-belchertown-new.
